Initial bootstrap

NaomiFPassarelli opened the pull request:
- Create initial bootstrap with yeoman basic structure and packages needed
- Added possibility to choose optional packages (carrousel, moment, etc)

Commit:
initial bootstrapp working

Commit:
add gitignore file

Commit:
change package.json info

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+  "name": "<%= projectNameSlugified %>",
+  "title": "<%= projectName %>",
+  "description": "<%= projectDescription %>",
+  "version": "0.0.0",
```
is it correct to have 0.0.0? I think we should set it to at least 0.1.0

mcallegari10 replied:
I agree, I'd use 1.0.0 instead

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+    "url": "<%= repoUrl %>"
+  },
+  "dependencies": {
+    <% if (includeAphrodite && !includeRadium) {%>
```
remove aphrodite

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+      "aphrodite": "^1.2.3",<%
+    }
+
+    if (!includeAphrodite && includeRadium) {%>
```
remove radium

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+    "i18next": "^9.0.0",<%
+    }
+
+    if (includeMobileDetect) {%>
```
I don't think this is necesary, do we have an example of a project where we are using it?

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+    "numeral": "^2.0.6",<%
+    }
+
+    if (includePostcss) {%>
```
we should add this by default, I would also add with it the autoprefixer(https://github.com/postcss/autoprefixer) and the sprites(https://github.com/2createStudio/postcss-sprites) plugins

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+    "react-modal": "^2.3.2",<%
+    }
+
+    if (includeReactResponsive) {%>
```
I would remove this from the selection. This is not used very often, I would use that library only  when we need it or directly define a utils function to determine a phone width and apply decitions based on that

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+    "react-google-maps": "^7.3.0",<%
+    }
+
+    if (includeReselect) {%>
```
include this by default

SKOLZ replied:
add this by default

SKOLZ commented on generators/app/templates/public/index.html:
```diff
@@ -0,0 +1,32 @@
+  <meta name="viewport" content="initial-scale=1">
+  <meta charset="utf-8">
+  
+  <!--
```
remove this comment

SKOLZ replied:
remove comment

SKOLZ commented on generators/app/templates/public/index.html:
```diff
@@ -0,0 +1,32 @@
+
+<body>
+  <div id="root"></div>
+  <!--
```
remove this comment

SKOLZ replied:
remove comment

Commit:
add i18next mandatory - remove aphrodite - update versions - add eslint - add inputValidations - separate reduxHandlers in actions and reducers

mcallegari10 commented on generators/app/templates/public/index.html:
```diff
@@ -0,0 +1,32 @@
+      The build step will place the bundled scripts into the <body> tag.
+      To begin the development, run `npm start`.
+      To create a production bundle, use `npm run build`.
+    -->
```
I'd delete this comment. Otherwise, it makes this a copy paste of create-react-app and we don't want that :smile:

mcallegari10 commented on generators/app/templates/src/app/components/Routes/components/AuthenticatedRoute.js:
```diff
@@ -0,0 +1,78 @@
+const DEFAULT_PUBLIC_ROUTE = Routes.LOGIN;
+const DEFAULT_PRIVATE_ROUTE = Routes.HOME;
+
+debugger;
```
![](http://www.reactiongifs.com/r/wsod.gif)

SKOLZ commented on generators/app/templates/_package.json:
```diff
@@ -0,0 +1,117 @@
+    "react-redux": "^5.0.6",
+    "react-router": "^4.2.0",
+    "react-router-dom": "^4.2.2",
+    "react-router-redux": "^5.0.0-alpha.6",
```
I would add this one if we are using redux or the react router 4 if we are not.

SKOLZ commented on generators/app/templates/src/app/components/Button/index.js:
```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+import Radium from "radium";
```
no radium, use css-modules instead

SKOLZ commented on generators/app/templates/eslintrc.js:
```diff
@@ -0,0 +1,47 @@
+    },
+    sourceType: 'module'
+  },
+  plugins: ['react', 'prettier', 'flowtype'],
```
we won't be using typed js

SKOLZ commented on generators/app/templates/eslintrc.js:
```diff
@@ -0,0 +1,47 @@
+    'airbnb',
+    'plugin:react/recommended',
+    'prettier',
+    'prettier/flowtype',
```
same

SKOLZ commented on generators/app/templates/src/app/components/Checkbox/index.js:
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import Radium from "radium";
```
change radium to css modules here too please

SKOLZ commented on generators/app/templates/src/app/components/Label/index.js:
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import Radium from "radium";
```
change radium to css modules here too please

SKOLZ commented on generators/app/templates/src/app/components/RadioButton/index.js:
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import Radium from "radium";
```
change radium to css modules here too please

Commit:
add files to gitignore - add reselector by default - clean code

pabloferro reviewed (dismissed):
Fix the package version and the debugger and 👍

Commit:
included postcss by default.

sbalay commented on generators/app/index.js:
```diff
@@ -0,0 +1,502 @@
+        store: true
+      },
+      {
+        type: "confirm",
```
I'd remove all these confirms and decide which libraries we want to always include and which libraries we don't.

When you kickoff a project you probably don't know if you will need recharts, carousel, mobileDetect, etc. Forcing a decision over something you don't know is more a problem than a solution.

sbalay commented on generators/app/index.js:
```diff
@@ -0,0 +1,502 @@
+"use strict";
```
this file is huge

sbalay commented on generators/app/index.js:
```diff
@@ -0,0 +1,502 @@
+          this[key] = answers[key];
+        }
+
+        function hasFeature(feat, prop) {
```
What's the goal of this function?

sbalay commented on generators/app/index.js:
```diff
@@ -0,0 +1,502 @@
+
+    mkdirp(this.destinationPath("src/app/assets/"));
+
+    this.fs.copyTpl(
```
Are all these files templates? Use `this.fs.copy` if they are not

sbalay commented on generators/app/index.js:
```diff
@@ -0,0 +1,502 @@
+    );
+
+    this.fs.copyTpl(
+      this.templatePath("public/index.html"),
```
We should be consistent with the naming of the templates. The template of `package.json` starts with `_`.  Use the same convention in all the files

sbalay reviewed (changes requested):
I guess this is enough for a first approach but we still need to run `create-react-app` here

Commit:
remove strict js - remove hasFeature func - change from copyTpl to copy - and _ to tmp

Commit:
add pull_request_template

add CRA

add complete CRA

change run - generator on the bottom

add prefix when run CRA

Commit:
clean run

Commit:
add curl and prerequisites in Readme

SKOLZ commented:
we decided to merge this to have something to work with
